scripts/basic-workflow-tests.sh

```diff
@@ -301,43 +351,55 @@ test_exec() {
 
     test_train
 
-    if [ "$FULLTRAIN" -eq 0 ]; then
-        # When we run training with --4-bit-quant, we can't convert the result to a gguf
-        # https://github.com/instructlab/instructlab/issues/579
-        # so we skip trying to test the result
-        return
+    # When we run training with --4-bit-quant, we can't convert the result to a gguf
+    # https://github.com/instructlab/instructlab/issues/579
+    # so we skip trying to test the result
+    if [ "$FULLTRAIN" -eq 1 ]; then
+        # When you run this --
+        #   `ilab model convert` is only implemented for macOS with M-series chips for now
+        #test_convert
+    
+        test_serve trained "${DATA_HOME}/instructlab/checkpoints/model.gguf"
+        PID=$!
+
+       test_chat
+
+        # Kill the serve process
+        task Stopping the ilab model serve for trained model
+        wait_for_server shutdown $PID
     fi
 
-    # When you run this --
-    #   `ilab model convert` is only implemented for macOS with M-series chips for now
-    #test_convert
-
-    test_serve "${DATA_HOME}/instructlab/models/ggml-model-f16.gguf"
-    PID=$!
-
-    test_chat
-
-    # Kill the serve process
-    task Stopping the ilab model serve
-    step Kill ilab model serve $PID
-    kill $PID
-
     task Evaluating the output of ilab model train
     test_evaluate
 }
 
-wait_for_server(){
-    if ! timeout 120 bash -c '
-        until curl -sS http://localhost:8000/docs &> /dev/null; do
-            echo "waiting for server to start"
-            sleep 1
+wait_for_server() {
+    ACTION="$1"
+    if [ "${ACTION}" == "start" ]; then
```

Comment:
(No action required) `==` is not POSIX compliant, it's usually better to use `=` which is identical and standard (but I understand the script is in `bash`.)